magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt

```diff
@@ -0,0 +1,56 @@
+package com.wealthfront.magellan.view
+
+import android.app.Dialog
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.wealthfront.magellan.lifecycle.LifecycleAware
+import javax.inject.Inject
+
+class DialogComponent @Inject constructor() : LifecycleAware {
+
+  private var dialogCreator: DialogCreator? = null
+  private var dialog: Dialog? = null
+  private var context: Context? = null
+
+  @VisibleForTesting internal var dialogIsShowing: Boolean = false
+
+  fun showDialog(dialogCreator: DialogCreator) {
+    this.dialogCreator = dialogCreator
+    this.dialogIsShowing = true
```

Comment:
Should we set this to false when the dialog is dismissed?

Reply:
d